build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -1097,31 +1097,31 @@ export const handlers: Handler[] = [
 					.join("")}`;
 			}
 
-			const clean = scripts["clean"];
-			if (clean && clean.startsWith("rimraf ")) {
-				if (clean.includes('"')) {
-					return "'clean' script using double quotes instead of single quotes";
+			if (cleanScript) {
+				if (cleanScript.includes("'")) {
+					return "'clean' script using single quotes instead of double quotes";
 				}
 
-				if (!clean.includes("'")) {
-					return "'clean' script rimraf argument should have single quotes";
+				if (!cleanScript.includes('"')) {
+					return "'clean' script rimraf glob arguments should have double quotes";
 				}
 			}
 		},
 		resolver: (file) => {
 			const result: { resolved: boolean; message?: string } = { resolved: true };
 			updatePackageJsonFile(path.dirname(file), (json) => {
 				const missing = missingCleanDirectories(json.scripts);
-				const clean = json.scripts["clean"] ?? "rimraf --glob";
-				if (clean.startsWith("rimraf --glob")) {
+				let clean: string = json.scripts["clean"] ?? "rimraf --glob";
+				if (!clean.startsWith("rimraf --glob")) {
 					result.resolved = false;
 					result.message =
 						"Unable to fix 'clean' script that doesn't start with 'rimraf --glob'";
-				}
-				if (missing.length === 0) {
 					return;
 				}
-				json.scripts["clean"] = `${clean} ${missing.map((name) => `'${name}'`).join(" ")}`;
+				if (missing.length !== 0) {
```

Comment:
nit: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v48.0.1/docs/rules/explicit-length-check.md

```suggestion
				if (missing.length > 0) {
```